factory: gh executable from PATH

wilso199 opened the pull request:
Changed cmd factory function to reference the `gh` executable from PATH.

Closes: #3704

Commit:
Using gh executable from PATH in favor of the resolved path

mislav committed:
Merge remote-tracking branch 'origin' into 3704-credential-helper

mislav committed:
Fix determining current process location

mislav reviewed (approved):
Thanks for taking a pass at this! The initial implementation didn't look alright to me, as I don't think `strings.HasSuffix(dir, "gh")` was a correct way to detect that `gh` executable was found at a particular point in PATH.

I've revised the implementation, improving support for relative entries in PATH as well as supplying reasonable fallback values. Please take a look!

wilso199 commented:
Looks great! Thanks for the help.

mislav committed:
Merge remote-tracking branch 'origin' into 3704-credential-helper

mislav committed:
Avoid resolving `executable()` until requested at runtime

This is to avoid hitting the filesystem and resolving symlinks
unnecessarily. The value of executable is just used conditionally by a
handful of commands.

mislav committed:
Ensure correct path to `gh` after `gh auth refresh` git credential setup

Militarysuperior3 commented:
One more load
